### DISCUSSION TITLE
Repo watching states are unclear

### DISCUSSION THREAD

**Discussion by Hyperseeker** [community/community#4107] [Other Feature Feedback, Questions, & Ideas]
At present, the way user-facing repo following is presented, it's unclear how to disable notifications from any particular repo.

But it's right there in the right-hand corner, right? Bear with me.

Some time ago, before the current version of the watching/following menu, I started following the releases for a particular repo. The repo was inactive, release-wise, and my hope was to get notified whenever work on it would resume.

At some point after that, the repo became active, release-wise. The problem is, it started releasing unstable versions of the app – something I wasn't interested in. However, any time I'd press the "Unsubscribe" button in the notifications list next to the release, it would not result in receiving no further notifications. On the notification page, my options were – as far as I was concerned – to say "Okay, I read it" or to keep it hanging. The "Unsubcribe" button, despite being there, seemed to do nothing.

When I went to the repo itself, in hopes of turning the notifications off first-hand, I was unable to locate the way to do so. I knew where the watching/following dropdown was, and I knew which button to press *before* the update, but the update brought confusion.

Because here's the thing:

What I wanted was to no longer receive notifications about releases.

I was unable to locate the exact button that would unsubscribe me from those. All I had was a bunch of states of following, whereas I wanted a "non-state" (the "non-state" of not following), something I couldn't easily locate.

The "Ignore" option is yet again a state, and the prompts me to "Stop ignoring" afterwards, something I wasn't after.

There is a "Custom" menu where I could specify which events to get notified about, but I cannot at present apply the "state" of not following any of the options. (If I untick all the boxes, I'm unable to press "Apply", which, in my mind, would result in complete unsubscription and return to the default.)

It took several attempts, once every few months when I got fed up again, to figure out that the default state is *not* "no notifications": it's "Participating and @mentions". I arrived to that result by sheer trial-and-error, with no other guidance provided.

Now. It's a minor detail, but when you have this sort of confusion in the system, it's alienating. If I can't rely on something as simple as the (un)subscription function, what other traps need I look out for? And there's a few: like the new `↓ Code` button which is supposed to replace the old separate "Clone"/"Download" couple, where each has a clear definition that the eyes of an unfamiliar or infrequent casual user could latch onto. (I say "new", even though it's been quite some time since the updates, because the "old" state is still present in my mind, after months and months with the new look. I hope that tells you something about how much weight it still has beyond the grumpy nostalgia.)

If I need to fork a repo, all I need to do is press the button clearly labeled "Fork" with the fork-y symbol next to the text. If I want to show my appreciation, I can press "Star": not exactly clear, but it's parsable as a verb. Then I get to the "Watch" and... if I'm not regularly browsing through repos for fun or for work, I might get lost. And this is basically the GUI for Git: shouldn't it take this sort of clarity into account?

My point is: the states of following/watching repos should be made clearer. The default state, whatever it is, should be labeled as such. Better still, there should be a button that lets you unsubscribe from all notifications and reset you to the default state.

I'm fine with it being the way it is; what I want is a clear way to get back to it from any position in the system.

**Comment by dchmelik** [comment]
I'm not fine with the way it is.  I shouldn't press 'follow' but then have to go to a menu then submenu within that just to follow releases and not just things I discuss (which I usually don't) because I'm following projects I like to compile/test when they have long-term/stable/user releases (and occasionally current/development/testing.)
&nbsp;&nbsp;&nbsp; I would like to follow distinct release typest, also.  I watch maybe 100+ projects (and will watch more in the future) but mostly only want to know about long-term/stable/user (normally-numbered) not current/development/testing (hexadecimal hash commit-numbered) releases.  I had to redirect all GitHub mail to a separate mailbox and just end up deleting most of it without reading.  Not allowing to watch certain types only causes a pretty big waste of time.
